android/app/src/main/java/org/electroncash/electroncash3/Main.kt

```diff
@@ -568,6 +583,55 @@ class WalletRenameDialog : AlertDialogFragment() {
     }
 }
 
+class WalletExportDialog : AlertDialogFragment() {
+    override fun onBuildDialog(builder: AlertDialog.Builder) {
+        builder.setTitle(R.string.export_wallet)
+                .setView(R.layout.wallet_export)
+                .setPositiveButton(android.R.string.ok, null)
+                .setNegativeButton(android.R.string.cancel, null)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): AlertDialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        dialog.window!!.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE)
+        return dialog
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onFirstShowDialog() {
+        val walletName = arguments!!.getString("walletName")!!
+        etExportFileName.setText(walletName)
+        etExportFileName.setSelection(0, etExportFileName.getText().length)
+    }
+
+    override fun onShowDialog() {
+        val walletName = arguments!!.getString("walletName")!!
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
+            val exportFileName = etExportFileName.text
+            if (exportFileName.contains('/')) {
+                toast(R.string.filenames_cannot)
+            } else {
+                val exportDir = File(activity!!.cacheDir, "wallet_exports")
+                exportDir.deleteRecursively() // To ensure no more than one temp file lingers
+                val exportFilePath = "$exportDir/$exportFileName"
+                val exportFile = File(exportFilePath)
+                val exportFileUri: Uri = FileProvider.getUriForFile(activity!!,
+                        "org.electroncash.wallet.wallet_exports", exportFile)
+                daemonModel.commands.callAttr("copy_wallet", walletName, exportFilePath)
+                val sendIntent = Intent()
+                sendIntent.type = "application/octet-stream"
+                sendIntent.action = Intent.ACTION_SEND
+                sendIntent.putExtra(Intent.EXTRA_STREAM, exportFileUri)
+                startActivity(Intent.createChooser(sendIntent, "SHARE"))
+                dismiss()
+            }
+        }
+    }
+}
+
+
+data class ShowSeedResult(val seed: String, val passphrase: String)
+
 
 data class SeedShowResult(val seed: String, val passphrase: String)
```

Comment:
Looks like the merge has left behind two copies of this class.

Reply:
Thanks for standardizing on NounVerb-style names, I agree it's an improvement. However, here I think we should remove the verb "Show" and just have "SeedResult" and "SeedPasswordDialog", because the current name makes it sound as if it's the password which is being shown, not the seed.

Reply:
Yeah, that did look a bit awkward.  Sorted now.